File: src/mango.rs
pub struct Mango {
    has_error: bool,
}

impl Mango {
    pub fn error(error: &str, file: &str, line: u32, line_contents: &str) {
        println!("{error} - {file}:{line}\n{line} | {line_contents}")
    }
}

File: src/mango.cpp
#include "mango.hpp"
#include "token.hpp"
#include <cstdint>
#include <iostream>
#include <string>

namespace Mango {
void error(std::wstring error, std::wstring file, std::uint32_t line) {
  std::wstring message = file;
  message += L":";
  message += std::to_wstring(line);
  message += L" | ";
  message += error;
  std::wcerr << error;
}

void error(std::wstring error) { std::wcerr << error; }

// TODO: make actually good errors
void error(Mango::TokenType type, std::wstring message) {
  if (type == TokenType::END_OF_FILE) {
    Mango::error(message + L" at end");
  } else {
    Mango::error(message);
  }
}
} // namespace Mango
